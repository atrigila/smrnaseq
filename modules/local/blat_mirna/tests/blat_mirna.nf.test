nextflow_process {

    name "Test Process BLAT_MIRNA"
    script "../blat_mirna.nf"
    process "BLAT_MIRNA"
    tag "modules"
    tag "modules_local"
    tag "blat_mirna"

    test("cDNA BLAT - Human") {

        when {
            params {
                outdir = "${outputDir}"
            }
            process {
                """
                input[0] = "cdna"
                input[1] =  [file("https://github.com/nf-core/test-datasets/raw/smrnaseq/miRBase/hairpin.fa", checkIfExists: true)]
                input[2] =  [file("https://huggingface.co/datasets/nf-core/smrnaseq/resolve/main/GRCh37/Homo_sapiens.GRCh37.cdna.all.fa", checkIfExists: true)]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()

            with(process.out.filtered_set) {
                with(get(0)) {
                    assert get(0).endsWith("filtered.fa")

                    // Check for AWK filtering of specific biotype
                    def lines = path(get(0)).readLines()
                    assert !lines.any { it.contains("transcript_biotype:miRNA") }

                    // Check for exclusion of miRNAs hits
                    assert !lines.any { it.contains("ENST00000564740.1") }
                }
            }
        }

    }

    test("ncRNA BLAT - Human") {

        when {
            params {
                outdir = "${outputDir}"
            }
            process {
                """
                input[0] = "ncRNA"
                input[1] =  [file("https://github.com/nf-core/test-datasets/raw/smrnaseq/miRBase/hairpin.fa", checkIfExists: true)]
                input[2] =  [file("https://huggingface.co/datasets/nf-core/smrnaseq/resolve/main/GRCh37/Homo_sapiens.GRCh37.ncrna.fa", checkIfExists: true)]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()

            with(process.out.filtered_set) {
                with(get(0)) {
                    assert get(0).endsWith("filtered.fa")

                    // Check for AWK filtering of specific biotype
                    def lines = path(get(0)).readLines()
                    // Lines contain transcript_biotype:rRNA or miRNA, so AWK assertions do not work:
                    //assert !lines.any { it.contains("transcript_biotype:rRNA") }
                    //assert !lines.any { it.contains("transcript_biotype:miRNA") }

                    // Check for exclusion of miRNAs hits
                    assert !lines.any { it.contains("ENST00000564740.1") }
                }
            }
        }

    }

    test("tRNA BLAT - Human") {

        when {
            params {
                outdir = "${outputDir}"
            }
            process {
                """
                input[0] = "tRNA"
                input[1] =  [file("https://github.com/nf-core/test-datasets/raw/smrnaseq/miRBase/hairpin.fa", checkIfExists: true)]
                input[2] =  [file("https://huggingface.co/datasets/nf-core/smrnaseq/resolve/main/GRCh37/hg19-tRNAs.fa", checkIfExists: true)]
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()

            with(process.out.filtered_set) {
                with(get(0)) {
                    assert get(0).endsWith("filtered.fa")

                    // Check for exclusion of miRNAs hits
                    def lines = path(get(0)).readLines()
                    assert !lines.any { it.contains("ENST00000564740.1") }
                }
            }
        }

    }

}
