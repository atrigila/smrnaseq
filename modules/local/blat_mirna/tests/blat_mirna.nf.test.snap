{
    "ncRNA BLAT - Human": {
        "content": [
            {
                "0": [
                    "filtered.fa:md5,6bc8a430400e2e78cf7f474981230811"
                ],
                "1": [
                    "versions.yml:md5,e2957df2cc8f0410101564c8e65d1761"
                ],
                "filtered_set": [
                    "filtered.fa:md5,6bc8a430400e2e78cf7f474981230811"
                ],
                "versions": [
                    "versions.yml:md5,e2957df2cc8f0410101564c8e65d1761"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-23T17:53:35.313580289"
    },
    "tRNA BLAT - Human": {
        "content": [
            {
                "0": [
                    "filtered.fa:md5,6b54e95ca5418d7d9c4d331ca3b2c96f"
                ],
                "1": [
                    "versions.yml:md5,e2957df2cc8f0410101564c8e65d1761"
                ],
                "filtered_set": [
                    "filtered.fa:md5,6b54e95ca5418d7d9c4d331ca3b2c96f"
                ],
                "versions": [
                    "versions.yml:md5,e2957df2cc8f0410101564c8e65d1761"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-23T17:57:58.238216453"
    },
    "cDNA BLAT - Human": {
        "content": [
            {
                "0": [
                    "filtered.fa:md5,8fd42894e815999b4278b08297720aae"
                ],
                "1": [
                    "versions.yml:md5,e2957df2cc8f0410101564c8e65d1761"
                ],
                "filtered_set": [
                    "filtered.fa:md5,8fd42894e815999b4278b08297720aae"
                ],
                "versions": [
                    "versions.yml:md5,e2957df2cc8f0410101564c8e65d1761"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-08-23T17:53:16.735132971"
    }
}